client/hosting/overview/components/migration-overview.tsx

```diff
@@ -0,0 +1,134 @@
+import { isFreePlanProduct } from '@automattic/calypso-products';
+import { translate } from 'i18n-calypso';
+import { HostingCard, HostingCardGrid } from 'calypso/components/hosting-card';
+import { HostingHero, HostingHeroButton } from 'calypso/components/hosting-hero';
+import { addQueryArgs } from 'calypso/lib/url';
+import { getMigrationStatus, getMigrationType } from 'calypso/sites-dashboard/utils';
+import type { SiteDetails } from '@automattic/data-stores';
+
+const cards = [
+	{
+		title: translate( 'Seriously secure' ),
+		text: translate(
+			'Firewalls, encryption, brute force, and DDoS protection. Your security’s all taken care of so you can stay one step ahead of any threats.'
+		),
+	},
+	{
+		title: translate( 'Unmetered bandwidth' ),
+		text: translate(
+			'With 99.999% uptime and entirely unmetered bandwidth and traffic on every plan, you’ll never need to worry about being too successful.'
+		),
+	},
+	{
+		title: translate( 'Power, meet performance' ),
+		text: translate(
+			'Our custom 28+ location CDN and 99.999% uptime ensure your site is always fast and always available from anywhere in the world.'
+		),
+	},
+	{
+		title: translate( 'Plugins, themes, and custom code' ),
+		text: translate(
+			'Build anything with full support and automatic updates for 50,000+ plugins and themes. Or start from scratch with your own custom code.'
+		),
+	},
+	{
+		title: translate( 'Expert support' ),
+		text: translate(
+			'Whenever you’re stuck, whatever you’re trying to make happen – our Happiness Engineers have the answers.'
+		),
+	},
+];
+
+const MigrationOverview = ( { site }: { site: SiteDetails } ) => {
+	if ( getMigrationStatus( site ) === 'pending' ) {
+		const migrationType = getMigrationType( site );
+
+		const isFreePlan = site.plan && isFreePlanProduct( site.plan );
+
+		const baseQueryArgs = {
+			siteId: site.ID,
+			siteSlug: site.slug,
+			start: 'true',
+			ref: 'hosting-migration-overview',
+		};
+
+		let continueMigrationUrl;
+
+		if ( isFreePlan ) {
+			// For the flows where the checkout is after the choice.
+			switch ( migrationType ) {
+				case 'diy':
+					continueMigrationUrl = addQueryArgs(
+						{
+							...baseQueryArgs,
+							destination: 'upgrade',
+							how: 'myself',
+						},
+						'/setup/site-migration/site-migration-upgrade-plan'
+					);
+					break;
+				case 'difm':
+					continueMigrationUrl = addQueryArgs(
+						{
+							...baseQueryArgs,
+							destination: 'upgrade',
+							how: 'difm',
+						},
+						'/setup/site-migration/site-migration-upgrade-plan'
+					);
+					break;
+				default:
+					continueMigrationUrl = addQueryArgs(
+						baseQueryArgs,
+						'/setup/site-migration/site-migration-how-to-migrate'
+					);
+			}
```

Comment:
Just a little concern, as we're only checking if the site is on a free plan or not, in case the site is on a `Personal` or `Premium` plan, will it send the user to the credentials or how-to-migrate step instead of the pre-checkout step?

Reply:
Hmm!! Good point, @Imran92! I was thinking that in our flows users wouldn't be able to pick a Personal or Premium plan and then try to migrate, but after you said, I tested it and I could do it through the `/start`, buying the plan (Personal or Premium), select the intent to migrate and use the flow.

I'll update it to a different check.

Reply:
Updated the check to match what we check in the flows to see if we'll redirect the user to the upgrade plan step or not. Now it makes more sense! Thank you for reviewing and catching this!

https://github.com/Automattic/wp-calypso/pull/95497/commits/3887d73cf58abd97025a2d745e6564fa5cc81d6d https://github.com/Automattic/wp-calypso/pull/95497/commits/2da505cadcab85fcba57cd756599270f2a7d7a18

Reply:
@renatho can you update the unit tests as well?

Reply:
Oops! Fixed here: https://github.com/Automattic/wp-calypso/pull/95497/commits/e5d856e258bcbb497c36a5665d59b4802c6261a9